packages/cli/lib/lib/webpack/render-html-plugin.js

```diff
@@ -54,9 +54,27 @@ module.exports = function(config) {
 		});
 	};
 
-	const pages = readJson(resolve(cwd, config.prerenderUrls || '')) || [
-		{ url: '/' },
-	];
+	let pages = [{ url: '/' }];
+
+	if (config.prerenderUrls) {
+		try {
+			let result = require(resolve(cwd, config.prerenderUrls));
+			if (typeof result.default !== 'undefined') {
+				result = result.default();
+			}
+			if (typeof result === 'function') {
+				result = result();
+			}
+			if (typeof result === 'string') {
+				result = JSON.parse(result);
+			}
+			if (result instanceof Array) {
+				pages = result;
+			}
+		} catch (error) {
+			warn(error.message);
```

Comment:
A custom error message along with it might be nice, not sure if the returned message will be obvious to users

Reply:
Yeah but that should be additional in that case. It'd be a shame to lose the extra information provided by the error message. In case of a syntax error it's easy to see what the syntax error was and in case of an incorrect path you can see the path that has been tried, which may have been resolved differently than a user would have expected.

Maybe: `Failed to load prerenderUrls file: ${error.message}`?

Reply:
yeah, that's what I meant

maybe even `error.stack`

Reply:
I'll play around a bit with different error messages when I get home.

Having vrijmibo right now :-)
<sub><sup>*Google it, should be sfw</sup></sub>

Reply:
Changed it to:
![screenshot 2019-01-21 at 10 39 30](https://user-images.githubusercontent.com/1649903/51465807-f7a37600-1d68-11e9-8d45-dd80a6f5daa9.png)